### PR TITLE
[Routing] updated RouteCollection::getUriForSite to preserve scheme from Site if it exists

### PR DESCRIPTION
**Pull request by eric-chau**
This PR aims to improve `RouteCollection::getUriForSite` to preserve HTTP scheme if it is provided by Site's `server_name`.

### CODE REVIEW

**Comment by eric-chau** [comment]
ping @crouillon =)

File: Routing/RouteCollection.php [hunk 1]
```diff
@@ -421,12 +421,13 @@ private function getUriForSite($pathinfo, Site $site = null)
             return $pathinfo;
         }
 
-        $protocol = $this->defaultScheme;
-        if (!empty($this->defaultScheme)) {
-            $protocol .= ':';
+        $protocol = (string) $this->defaultScheme;
+        $protocol = $protocol . (false != $protocol ? ':' : '') . '//';
+        if ($site instanceof Site && null !== parse_url($site->getServerName(), PHP_URL_SCHEME)) {
```
Comment (eric-chau) [inline Routing/RouteCollection.php]: Thank you @crouillon for your feedbacks. I took in account your comments and updated `::getUriForSite`. Please check it once again when you can.

**Comment by crouillon** [comment]
👍 Thanks @eric-chau

**Comment by eric-chau** [comment]
thank you @crouillon for your code review

**Pull request merged**